subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -717,6 +718,122 @@ const struct lwm2m_reader oma_tlv_reader = {
 	get_opaque
 };
 
+int do_read_op_tlv(struct lwm2m_engine_obj *obj,
```

Comment:
Yeah, at the end of all the work, they do seem very similar.  The outlier is plain-text where you have almost no processing at all since it only handles single results.

Let me see what I can do to cut down on duplicate code there.